### PR TITLE
PM-539 fix the default retention policy name

### PR DESCRIPTION
**Pull request by ramazanyich**

### CODE REVIEW

**Review by francisdb — COMMENTED** [review]
Since what version of influx did this change?
Is this not tested or does it silently fail?

**Comment by ramazanyich** [comment]
I see it at least with 1.8 influxdb. 
Actually we try to adjust it during db creation. But it returns 200 HTTP code.

**Comment by ramazanyich** [comment]
> I see it at least with 1.8 influxdb. Actually we try to adjust it during db creation. But it returns 200 HTTP code.

looked in history of influxDB documentation. The change was done between version 0.13 and 1.0 
from 0.13 docs:

"""
When you create a database, InfluxDB automatically creates an RP called **default** with an infinite duration and a replication factor set to the number of nodes in the cluster. default also serves as the DEFAULT RP; if you do not supply an explicit RP when you write a point to the database, the data is subject to the DEFAULT RP.
"""

and from 1.0 docs:
"""
When we created the food_data database in step 1, InfluxDB automatically generated an RP named **autogen** and set it as the DEFAULT RP for the database. The autogen RP has an infinite retention period. 
"""

**Comment by ramazanyich** [comment]
Should not we create a retention policy together with db creation? it is possible to do in one line

**Comment by francisdb** [comment]
https://github.com/influxdata/influxdb/issues/3733
and https://github.com/influxdata/influxdb/commit/baaa782c9504323d054841ae10e82b9543d6351a
which is indeed introduced in [1.8.0](https://github.com/influxdata/influxdb/releases/tag/1.8.0)

I wonder if we can query the settings to find out what the name is

**Review by francisdb — APPROVED** [review]
Let's clearly document this is for influx >= 1.8.0 in the release notes

**Comment by francisdb** [comment]
indeed let's do it in a single step

**Comment by francisdb** [comment]
> and from 1.0 docs:

Are you sure these are not 1.x docs?

**Comment by ramazanyich** [comment]
> > and from 1.0 docs:
> 
> Are you sure these are not 1.x docs?

here are the links: 
https://archive.docs.influxdata.com/influxdb/v0.13/guides/downsampling_and_retention/ - it mentions **default** retention policy
https://archive.docs.influxdata.com/influxdb/v1.0/guides/downsampling_and_retention/ - it mentions **autogen** retention policy

**Comment by francisdb** [comment]
Yes you are correct, looks like the github tag lister works in mysterious ways, clicking `...` shows 1.0.0
https://github.com/influxdata/influxdb/commit/baaa782c9504323d054841ae10e82b9543d6351a

File: src/main/scala/io/waylay/influxdb/query/QueryResultProtocol.scala [hunk 1]
```diff
@@ -27,7 +27,14 @@ object QueryResultProtocol {
     case other  => fieldValueReads.reads(other).map(Some(_))
   }
 
-  implicit val serieReads: Reads[Serie]     = Json.reads[Serie]
+  implicit val serieReads: Reads[Serie] = jsValue =>
```
Comment (francisdb) [inline src/main/scala/io/waylay/influxdb/query/QueryResultProtocol.scala:30]: any reason for this? Not that I am against it.

**Review by francisdb — APPROVED** [review]

File: src/main/scala/io/waylay/influxdb/query/QueryResultProtocol.scala (hunk 1 above)
Comment (ramazanyich) [reply src/main/scala/io/waylay/influxdb/query/QueryResultProtocol.scala:30]: response for SHOW RETENTION POLICIES does not have name field.

Comment (ramazanyich) [reply src/main/scala/io/waylay/influxdb/query/QueryResultProtocol.scala:30]: added separate unit test for reading.

**Review by brunoballekens — APPROVED** [review]

**Pull request merged**